src/Plugins/ComplexCore/src/ComplexCore/Filters/FindBoundaryCellsFilter.cpp

```diff
@@ -0,0 +1,122 @@
+IFilter::UniquePointer FindBoundaryCellsFilter::clone() const
+{
+  return std::make_unique<FindBoundaryCellsFilter>();
+}
+
+//------------------------------------------------------------------------------
+IFilter::PreflightResult FindBoundaryCellsFilter::preflightImpl(const DataStructure& dataStructure, const Arguments& filterArgs, const MessageHandler& messageHandler,
+                                                                const std::atomic_bool& shouldCancel) const
+{
+  auto pImageGeometryPath = filterArgs.value<DataPath>(k_GeometryPath_Key);
+  auto pFeatureIdsArrayPathValue = filterArgs.value<DataPath>(k_FeatureIdsArrayPath_Key);
+  auto pBoundaryCellsArrayNameValue = filterArgs.value<std::string>(k_BoundaryCellsArrayName_Key);
+
+  const DataPath boundaryCellsArrayPath = pFeatureIdsArrayPathValue.getParent().createChildPath(pBoundaryCellsArrayNameValue);
+
+  PreflightResult preflightResult;
+  complex::Result<OutputActions> resultOutputActions;
+  std::vector<PreflightValue> preflightUpdatedValues;
+
+  const auto& featureIds = dataStructure.getDataRefAs<Int32Array>(pFeatureIdsArrayPathValue);
+  const usize numFeatureIds = featureIds.getSize();
+  const usize numCells = dataStructure.getDataRefAs<ImageGeom>(pImageGeometryPath).getNumberOfCells();
+  if(numFeatureIds != numCells)
+  {
+    return {MakeErrorResult<OutputActions>(
+        -5320,
+        fmt::format("The selected Image geometry '{}' and feature ids array '{}' have mismatching number of elements. Make sure the selected feature ids was created for the selected Image geometry.",
+                    pImageGeometryPath.toString(), pFeatureIdsArrayPathValue.toString()))};
+  }
+
+  auto action = std::make_unique<CreateArrayAction>(DataType::int8, featureIds.getTupleShape(), std::vector<usize>{1}, boundaryCellsArrayPath);
+  resultOutputActions.value().actions.push_back(std::move(action));
+
+  return {std::move(resultOutputActions), std::move(preflightUpdatedValues)};
+}
+
+//------------------------------------------------------------------------------
+Result<> FindBoundaryCellsFilter::executeImpl(DataStructure& dataStructure, const Arguments& filterArgs, const PipelineFilter* pipelineNode, const MessageHandler& messageHandler,
+                                              const std::atomic_bool& shouldCancel) const
+{
+  FindBoundaryCellsInputValues inputValues;
+
+  inputValues.IgnoreFeatureZero = filterArgs.value<bool>(k_IgnoreFeatureZero_Key);
+  inputValues.IncludeVolumeBoundary = filterArgs.value<bool>(k_IncludeVolumeBoundary_Key);
+  inputValues.ImageGeometryPath = filterArgs.value<DataPath>(k_GeometryPath_Key);
+  inputValues.FeatureIdsArrayPath = filterArgs.value<DataPath>(k_FeatureIdsArrayPath_Key);
+  const auto pBoundaryCellsArrayNameValue = filterArgs.value<std::string>(k_BoundaryCellsArrayName_Key);
+  inputValues.BoundaryCellsArrayName = inputValues.FeatureIdsArrayPath.getParent().createChildPath(pBoundaryCellsArrayNameValue);
```

Comment:
```suggestion
  inputValues.BoundaryCellsArrayName = inputValues.FeatureIdsArrayPath.getParent().createChildPath(filterArgs.value<std::string>(k_BoundaryCellsArrayName_Key));
```
OPTIONAL: since we are only using the value once just inline it to avoid unnecessary copy